{
  try {
    listener.get();
    fail(msg);
  }
 catch (  ExecutionException ex) {
    assertThat(ex.getCause(),instanceOf(klass));
  }
}
